e2e/replay/msg_middleware.go

```diff
@@ -0,0 +1,105 @@
+package replay
+
+import (
+	"log"
+	"time"
+
+	"github.com/0xPolygon/pbft-consensus"
+	"github.com/0xPolygon/pbft-consensus/e2e"
+)
+
+const (
+	FileName = "messages"
+)
+
+// MessagesNotifier is a struct that implements ReplayNotifier interface
+type MessagesNotifier struct {
```

Comment:
I was thinking about renaming this struct into `MessagesMiddleware`, as it represents more than just a "notifier". WDYT?

Reply:
Agree, done